python/nnvm/compiler/build_module.py

```diff
@@ -156,11 +156,13 @@ def optimize(graph, shape, dtype="float32"):
     cfg = BuildConfig.current
     if cfg.pass_enabled("SimplifyInference"):
         graph = graph_attr.set_shape_inputs(graph, shape)
-        graph = graph.apply(["InferShape", "SimplifyInference"])
+        graph = graph.apply(["InferShape", "ExpandCompute",
+                             "InferShape", "SimplifyInference"])
```

Comment:
it'd be nice to mark a pass as preserving an attribute so shape doesn't have to be inferred so often

Reply:
We are working on something toward that